Arm/State.lean

```diff
@@ -290,6 +290,45 @@ def r (fld : StateField) (s : ArmState) : (state_value fld) :=
   | FLAG i  => read_base_flag i s
   | ERR     => read_base_error s
 
+@[state_simp_rules]
+def ArmState.x0 (s : ArmState) : BitVec 64 := r (StateField.GPR 0) s
+
+@[state_simp_rules]
+def ArmState.x1 (s : ArmState) : BitVec 64 := r (StateField.GPR 1) s
+
+@[state_simp_rules]
+def ArmState.x2 (s : ArmState) : BitVec 64 := r (StateField.GPR 2) s
+
+@[state_simp_rules]
+def ArmState.sp (s : ArmState) : BitVec 64 := r (StateField.GPR 31) s
+
+@[state_simp_rules]
+def ArmState.V (s : ArmState) : BitVec 1 := r (StateField.FLAG PFlag.V) s
+
+@[state_simp_rules]
+def ArmState.C (s : ArmState) : BitVec 1 := r (StateField.FLAG PFlag.C) s
+
+@[state_simp_rules]
+def ArmState.Z (s : ArmState) : BitVec 1 := r (StateField.FLAG PFlag.Z) s
+
+@[state_simp_rules]
+def ArmState.N (s : ArmState) : BitVec 1 := r (StateField.FLAG PFlag.N) s
+
+def ArmState.r_GPR_0_eq_x0 (s : ArmState) : r (StateField.GPR 0) s = s.x0 := by rfl
```

Comment:
It's not immediate clear what these definitions are good for. Maybe consider adding a comment.

Reply:
These dot notations are convenient, but there are more than 4 gprs and also sfps. I'm wondering if there is a better way to do this than listing them out.

Reply:
@pennyannn I don't immediately know of a better way that is as simple as rolling our sleeves and listing them out :) Metaprogramming games can be played, but my feeling is that the complexity isn't worth it.

Reply:
Understandable.

Reply:
These should probably be `abbrev`s